ksql-cli/src/main/java/io/confluent/ksql/cli/console/JLineReader.java

```diff
@@ -52,40 +52,43 @@
    */
   private static class KsqlExpander extends DefaultExpander {
 
-    private static final Map<String, String> shortcuts = Maps.newHashMap();
-    
-    {
-      shortcuts.put("cs", "CREATE STREAM s ( field1 type1 ) WITH (KAFKA_TOPIC='topic',"
-          + " VALUE_FORMAT='avro');");
-      shortcuts.put("ct", "CREATE TABLE t ( field1 type1 ) WITH (KAFKA_TOPIC='topic',"
-          + " VALUE_FORMAT='avro', KEY='field1');");
-      shortcuts.put("csas", "CREATE STREAM s AS SELECT ");
-      shortcuts.put("ctas", "CREATE TABLE t AS SELECT ");
-      shortcuts.put("ii", "INSERT INTO x SELECT ");
-    }
+    private static final String EXPANDED_CS =
+        "CREATE STREAM s (field1 type1, field2 type2) "
+            + "WITH (KAFKA_TOPIC='topic-name', VALUE_FORMAT='json');";
+
+    private static final String EXPANDED_CT =
+        "CREATE TABLE t (field1 type1, field2 type2) "
+            + "WITH (KAFKA_TOPIC='topic-name', VALUE_FORMAT='json', KEY='field1');";
+
+    private static final Map<String, String> shortcuts = ImmutableMap.of(
+        "cs", EXPANDED_CS,
+        "ct", EXPANDED_CT,
+        "csas", "CREATE STREAM s AS SELECT ",
+        "ctas", "CREATE TABLE t AS SELECT ",
+        "ii", "INSERT INTO x SELECT "
+    );
 
     @Override
     public String expandHistory(final History history, final String line) {
       if (line.startsWith("!") || line.startsWith("^")) {
         return super.expandHistory(history, line);
-      } else {
-        return line;
       }
+
+      return line;
     }
 
     @Override
     public String expandVar(final String word) {
-      final String snippet = shortcuts.getOrDefault(word.toLowerCase(), word);
-      return snippet;
+      return shortcuts.getOrDefault(word.toLowerCase(), word);
     }
-
   }
 
   JLineReader(final Terminal terminal, final Path historyFilePath) {
     // The combination of parser/expander here allow for multiple-line commands connected by '\\'
     final DefaultParser parser = new DefaultParser();
     parser.setEofOnEscapedNewLine(true);
-    parser.setQuoteChars(new char[0]);
+    parser.setEofOnUnclosedQuote(true);
```

Comment:
This enables JLine to not accept a line with unclosed quotes, i.e. it stops the CLI terminating a line due to a `;` if its in quotes, e.g. ensures this:

```
SELECT 'some weird value that just happens to end in a ;
 and is spread over several lines' AS something FROM X;
```

Is handled correctly.

Reply:
Going through the JLine docs to find the right settings must have been fun! ;-)

Reply:
What docs? ;)